build.py

```diff
@@ -131,26 +154,44 @@ def deploy(self, service):
         logging.info("Upload Docker image from service %s to Docker hub." % service)
         subprocess.check_call("docker-compose push %s" % service, shell=True)
 
+    def info(self, service):
+        """Show Docker image info
+        :param service: Docker compose service name
+        """
+        image = self._get_image_name(service)
+        logging.info("Show Docker image %s size:" % image)
+        subprocess.call('docker images %s' % image, shell=True)
+        logging.info("Show Docker image %s info:" % image)
+        subprocess.call('docker inspect %s' % image, shell=True)
+
     def run(self):
         """Execute all 3 stages for all versions in compilers list
         """
         for arch in self.variables.docker_archs:
-            for compiler in [self.gcc_compiler, self.clang_compiler]:
+            for compiler in [self.gcc_compiler, self.clang_compiler, self.visual_compiler]:
                 for version in compiler.versions:
                     tag_arch = "" if arch == "x86_64" else "-%s" % arch
                     service = "%s%s%s" % (compiler.name, version.replace(".", ""), tag_arch)
                     build_dir = "%s_%s%s" % (compiler.name, version, tag_arch)
 
-                    self.linter(build_dir)
+                    if platform.system() == "Linux":
+                        self.linter(build_dir)
                     self.build(service)
-                    self.test(arch, compiler.name, version, service)
-                    self.deploy(service)
+                    try:
+                        float(version) # base versions will not be tested and deployed
```

Comment:
This is a bit tricky. If the base is always "base" I would prefer a comparison with a string or a different approach.

Reply:
I agree!

Reply:
removed